backend/experiment/rules/matching_pairs.py

```diff
@@ -117,14 +118,42 @@ def get_matching_pairs_trial(self, session):
             title='Tune twins',
             playback=playback,
             feedback_form=None,
-            result_id=prepare_result('matching_pairs', session),
+            result_id=prepare_result('matching_pairs_board', session),
             config={'show_continue_button': False}
         )
         return trial
 
     def calculate_score(self, result, data):
-        moves = data.get('result').get('moves')
-        for m in moves:
-            m['filename'] = str(Section.objects.get(pk=m.get('selectedSection')).filename)
-        score = data.get('result').get('score')
-        return score
\ No newline at end of file
+        ''' in this experiment, this saves only the cleared board '''
+        moves = list(result.session.result_set.filter(question_key='move'))
+        scores = sum(m.score for m in moves)
+        return 100 + scores
+    
+    def calculate_intermediate_score(self, session, result):
+        ''' will be called every time two cards have been turned '''
+        result_data = json.loads(result)
+        first_card = result_data['lastCard']
+        first_section = Section.objects.get(pk=first_card['id'])
+        first_card['filename'] = str(first_section.filename)
+        second_card = result_data['currentCard']
+        second_section = Section.objects.get(pk=second_card['id'])
+        second_card['filename'] = str(second_section.filename)
+        if first_section.group == second_section.group:
+            if 'seen' in second_card:
+                score = 20
+                given_response = 'match'
+            else:
+                score = 10
+                given_response = 'lucky match'
+        else:
+            if 'seen' in second_card:
+                score = -10
+                given_response = 'misremembered'
+            else:
+                score = 0
+                given_response = 'no match'
+        prepare_result('move', session, json_data=result_data,
+                       score=score, given_response=given_response)
+        return score
```

Comment:
Perhaps a unit test can be added for this method that simulates the requests made from the frontend.